packages/asset-swapper/src/utils/quote_requestor.ts

```diff
@@ -121,10 +117,16 @@ export class QuoteRequestor {
             'firm',
         );
 
+        const ordersWithStringInts = firmQuotes.map(quote => quote.signedOrder);
+
         const validatedOrdersWithStringInts = ordersWithStringInts.filter(order => {
-            const hasValidSchema = this._schemaValidator.isValid(order, schemas.signedOrderSchema);
-            if (!hasValidSchema) {
-                this._warningLogger(order, 'Invalid RFQ-t order received, filtering out');
+            try {
+                const hasValidSchema = this._schemaValidator.isValid(order, schemas.signedOrderSchema);
+                if (!hasValidSchema) {
+                    throw new Error('order not valid');
+                }
+            } catch (err) {
+                this._warningLogger(order, `Invalid RFQ-t order received, filtering out. ${err.message}`);
```

Comment:
When would this throw instead of `hasValidSchema` just returning `false`?

Reply:
When the input isn't even valid JSON.  Returning false implies that the response is valid JSON but doesn't conform to the schema.  See test case I added for this in the commit that introduced this change: https://github.com/0xProject/0x-monorepo/pull/2582/commits/b9984b6df4555f4d4417d0589dff2f4214ad76a6

Reply:
Got it, makes sense -- thanks!